### PR TITLE
fix: test case `ut_lind_fs_getdents`

### PR DESCRIPTION
**Pull request by ChinmayShringi**
## Description

Fixes # (issue)

This pull request focuses on resolving alignment issues in the `ut_lind_fs_getdents` test and ensuring proper handling of directory entries when interacting with packed fields. The primary changes involve:
- **Ensuring Safe Access to Packed Fields**: In the original test, direct access to packed fields in the `ClippedDirent` struct resulted in alignment issues, particularly for `d_off` and `d_reclen`. These fields have been safely accessed by copying them into local variables to ensure proper memory alignment.
- **Test Setup Cleanup**: The test now ensures that the `/getdents` directory is properly cleaned up before the test is executed. If the directory already exists, it is removed to avoid conflicts. This ensures the test runs in a clean environment each time.

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Dependencies
This change does not introduce new dependencies, but it ensures better integration with existing modules (e.g., `native-client`, `lind-glibc`, and `lind-project`). No updates are required in those modules for this change to be effective.

## How Has This Been Tested?

- `cargo test ut_lind_fs_getdents`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, lind-project)

### CODE REVIEW

File: src/tests/fs_tests.rs [hunk 1]
```diff
@@ -2512,26 +2512,38 @@ pub mod fs_tests {
...
+        // Check the return value of getdents_syscall
         assert_eq!(cage.getdents_syscall(fd, baseptr, bufsize as u32), 48);
+        let result = cage.getdents_syscall(fd, baseptr, bufsize as u32);
+        assert!(result >= 0, "getdents_syscall failed with error: {}", result);
 
         unsafe {
             let first_dirent = baseptr as *mut interface::ClippedDirent;
-            assert!((*first_dirent).d_off == 24);
-            let reclen_matched: bool = ((*first_dirent).d_reclen == 24);
+            // Copy packed fields into local variables to avoid alignment issues
```
Comment (JustinCappos) [inline src/tests/fs_tests.rs]: Can you explain more what this is doing?   What alignment?   Do you mean word, etc. alignment?   Why is this needed?

File: src/tests/fs_tests.rs [hunk 2]
```diff
@@ -2540,6 +2552,7 @@ pub mod fs_tests {
         assert_eq!(cage.exit_syscall(libc::EXIT_SUCCESS), libc::EXIT_SUCCESS);
         lindrustfinalize();
     }
+
```
Comment (JustinCappos) [inline src/tests/fs_tests.rs:2569]: extra space add?

Comment (ChinmayShringi) [reply src/tests/fs_tests.rs:2569]: Yes, I added a new line between the 2 test functions

File: src/tests/fs_tests.rs (hunk 1 above)
Comment (ChinmayShringi) [reply src/tests/fs_tests.rs]: This is a byte alignment issue, the fields in the packed structure (ClippedDirent) are tightly packed without the usual padding that would align them on word boundaries (like 4 or 8 bytes). Because of this, directly accessing these fields cause problems on some systems that expect memory to be accessed in larger chunks (like words).

Comment (JustinCappos) [reply src/tests/fs_tests.rs]: Can you explain this in the comment in more detail?   Once this happens, I'll approve

**Review by Yaxuan-w — APPROVED** [review]
I worked with Qianxi on testing `getdents` before, and we encountered issues related to memory alignment and differences in the data structures between 32-bit and 64-bit layouts when checking the returned structure. I'm really glad to see that your test is comprehensive in addressing these concerns.
